pkg/apis/serving/k8s_validation.go

```diff
@@ -418,7 +419,8 @@ func validateContainerPorts(ports []corev1.ContainerPort) *apis.FieldError {
 		userPort.ContainerPort == networking.BackendHTTP2Port ||
 		userPort.ContainerPort == networking.QueueAdminPort ||
 		userPort.ContainerPort == networking.AutoscalingQueueMetricsPort ||
-		userPort.ContainerPort == networking.UserQueueMetricsPort {
+		userPort.ContainerPort == networking.UserQueueMetricsPort ||
+		userPort.ContainerPort == pkgNetworking.ProfilingPort {
```

Comment:
I was looking at https://github.com/knative/serving/blob/master/docs/runtime-contract.md and there's nothing about ports in queue-proxy. If I understand it correctly the contract is mostly for the user container.
Anyway, it might be good to add a whole section regarding allowed port range and ports that are reserved for queue-proxy, including 8012,8013,8022,9090,9091 and now 8008 as well.